src/components/listContext.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+import { getComments } from './todosApi/api-todos';
+import { Comment } from '../types/Comment';
+
+type ListContextType = {
+  users: User[];
+  setUsers: Dispatch<SetStateAction<User[]>>;
+  setIsLoader: Dispatch<SetStateAction<boolean>>;
+  isLoader: boolean;
+  setErrorPosts: Dispatch<SetStateAction<boolean>>;
+  errorPosts: boolean;
+  post: Post[];
+  setPost: Dispatch<SetStateAction<Post[]>>;
+  setDetail: Dispatch<SetStateAction<boolean>>;
+  detail: boolean;
+  fetchUserComments: (postId: number) => void;
+  comments: Comment[];
+  errorComments: boolean;
+  setErrorComments: Dispatch<SetStateAction<boolean>>;
+  selectedUser: User | null;
+  setSelectedUser: Dispatch<SetStateAction<User | null>>;
+  selectedPostId: number | null;
+  setSelectedPostId: Dispatch<SetStateAction<number | null>>;
+  loaderDetails: boolean;
+  setIsLoaderDetails: Dispatch<SetStateAction<boolean>>;
+  buttonAddComment: boolean;
+  setButtonAddComment: Dispatch<SetStateAction<boolean>>;
+  addComment: boolean;
```

Comment:
```suggestion
  isAddCommentFormVisible: boolean;
```
long but meaningful name always better than short